examples/front.kk

```diff
@@ -0,0 +1,97 @@
+/*----------------------------------------------------------------------------
+   Copyright 2024, Koka-Community Authors
+
+   Licensed under the MIT License ("The License"). You may not
+   use this file except in compliance with the License. A copy of the License
+   can be found in the LICENSE file at the root of this distribution.
+----------------------------------------------------------------------------*/
+
+import html/html
+
+import html/layout
+
+// todo: this deserves a PR to go into std/core?
+fun do(action : () -> e a) : e a
+  action()
+
+value struct cnode
```

Comment:
Koka warns: Type cnode requires 40 bytes which is rather large for a value type

does this mean `value` keyword will force the generated machine code pass that struct values via registers?

i wonder if `string` fields would also have its var-len payload packed altogether, i'm think about https://github.com/koka-lang/koka/issues/543 , mini-heaps with var-len data seem to work performantly that way too.

Reply:
Koka ultimately leaves the registers / stack up to the C compiler. However, it does mean that the string struct will be embedded in the `cnode` struct instead of having a pointer to that value. However, it doesn't change whether the strings themselves are value / ref objects. Strings are currently represented by a length and then the pointer to the chars (not necessarily allocated next to the length field).